velox/functions/sparksql/String.h

```diff
@@ -1130,4 +1130,52 @@ struct ReplaceFunction {
   }
 };
 
+template <typename T>
+struct FindInSetFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& strArray) {
+    int32_t index = 1;
+    int32_t lastComma = -1;
+
+    if (std::string_view(str).find(',') != std::string::npos) {
+      result = 0;
+      return;
+    }
+
+    auto arrayData = strArray.data();
+    size_t arraySize = strArray.size();
+    size_t matchSize = str.size();
+
+    for (int i = 0; i < arraySize; i++) {
```

Comment:
One possible optimization.
For constant `strArray`, in `initialize()` (see [link](https://github.com/facebookincubator/velox/blob/main/velox/docs/develop/scalar-functions.rst#access-to-session-properties-and-constant-inputs)), I think we can split it with comma delimiter and create a hash map to hold unique elements and their indexes. Then, in `call()`, for such constant case, we can just do a search from hash map for each row. Does it make sense?

For non-constant case, I think we can keep the current handling unchanged in `call`.

Reply:
Thanks, I think both `str` and `strArray` might be a different value, i think that we can keep our current processing logic.

Reply:
Not sure how common column `strArrray` is used.
Hi @mbasmanova, I have reviewed this pr. Do you have any comment on this or other part? Thanks!